src/esm_runscripts/echam.py

```diff
@@ -0,0 +1,176 @@
+        nproma = 8
+        lcouple = .true.
+        getocean = 1, 'days', 'last', 0
+        putocean = 1, 'days', 'last', 0
+        lcouple_co2 = .true.
+        default_output = .false.
+    /
+
+    &parctl
+        nproca = 24
+        nprocb = 24
+    /
+
+    &submodelctl
+        lmethox = .true.
+    /
+
+    &submdiagctl
+        vphysc_lpost = .false.
+    /
+
+    &radctl
+        iaero = 3
+        io3 = 4
+        isolrad = 6
+        ich4 = 3
+        in2o = 3
+        co2vmr = 284.3169860840e-06
+        ch4vmr = 808.2490234375e-09
+        n2ovmr = 273.0210571289e-09
+        yr_perp = 1850
+    /
+    &mvstreamctl
+        filetag = 'paul_custom'
+        source = 'g3b'
+        variables = 'temp2:mean>temp2=167'
+        interval = 1, 'months', 'last', 0
+    /
+
+    The following code will extract the mvstream tags from the namelist:
+
+    >>> namelist_path = "/tmp/example/namelist.echam"
+    >>> tags = _get_mvstream_tags_from_namelist(namelist_path)
+    >>> print(tags)
+    ['paul_custom']
+    """
+    mvstream_tags = []
+    if not isinstance(namelist, f90nml.namelist.Namelist):
+        try:
+            namelist = f90nml.read(namelist)
+        except FileNotFoundError:
+            logger.error(f"Namelist specified by {namelist} could not be found")
+        except TypeError as e:
+            logger.error(
+                f"Could not convert {namelist} to f90nml.namelist.Namelist object."
+            )
+            raise e
+
+    for chapter, contents in namelist.items():
+        if chapter == "mvstreamctl":
+            tag = contents.get("filetag")
+            if tag is not None:
+                mvstream_tags.append(tag)
+    return mvstream_tags
+
+
+def append_namelist_dependent_sources(config):
+    """
+    Append namelist dependent sources to the ECHAM configuration.
+
+    This function updates the `outdata_sources` in the ECHAM configuration
+    based on the namelist objects and other configuration parameters.
+
+    Parameters
+    ----------
+    config : dict
+        The configuration dictionary containing general, ECHAM, and JSBACH settings.
+
+    Notes
+    -----
+    - The function reads the namelist from the specified directory if not
+      already loaded.
+    - It filters out tags that are to be ignored based on the JSBACH streams
+      or specified ignore tags.
+    - The output file type is checked, and if it is NetCDF (indicated by
+      ``out_filetype`` == 2), the file extension `.nc` is appended to the tags.
+    - The function logs the updates made to the ``outdata_sources``.
+    """
+    expid = config["general"]["expid"]
+    econfig = config["echam"]
+    try:
+        namelist = econfig["namelist_objs"]
+    except KeyError:  # Namelists not yet loaded...
+        namelist = f90nml.read(econfig["namelist_dir"] + "/namelist.echam")
+    mvstream_tags = _get_mvstream_tags_from_namelist(namelist)
+    jsbach_streams = config["jsbach"]["streams"]
```

Comment:
```suggestion
    jsbach_streams = config["jsbach"].get("streams", [])
```
Isn't it better/safer to use a get here?

Reply:
yes, that is safer, the user *might* have the silly idea to remove the default JSBACH streams.